src/librustc_codegen_llvm/back/write.rs

```diff
@@ -239,9 +239,7 @@ impl<'a> Drop for DiagnosticHandlers<'a> {
     }
 }
 
-unsafe extern "C" fn report_inline_asm(cgcx: &CodegenContext<LlvmCodegenBackend>,
-                                       msg: &str,
-                                       cookie: c_uint) {
+unsafe fn report_inline_asm(cgcx: &CodegenContext<LlvmCodegenBackend>, msg: &str, cookie: c_uint) {
     cgcx.diag_emitter.inline_asm_error(cookie as u32, msg.to_owned());
```

Comment:
Is this method `unsafe`? Doesn't seem to me like there would be a reason to mark `report_inline_asm` as `unsafe`.

Reply:
Pretty sure it doesn't. Doesn't have to happen in this PR, but since it's being touched anyway, let's remove the `unsafe` too.

Reply:
I've removed the `unsafe`.